modules/node-shared/src/main/scala/org/tessellation/node/shared/infrastructure/snapshot/CurrencySnapshotAcceptanceManager.scala

```diff
@@ -166,6 +178,35 @@ object CurrencySnapshotAcceptanceManager {
           .getOrElse(acc)
       }
 
+    private def acceptFeeTxs(
+      balances: SortedMap[Address, Balance],
+      maybeTxs: Option[SortedSet[Signed[FeeTransaction]]]
+    ): F[(SortedMap[Address, Balance], Option[SortedSet[Signed[FeeTransaction]]])] =
+      maybeTxs match {
+        case None => (balances, maybeTxs).pure[F]
+        case Some(txs) =>
+          txs
+            .foldLeft(balances.view.mapValues(_.value.value).toMap) {
+              case (balances, tx) =>
+                val src = balances.getOrElse(tx.source, Balance.empty.value.value) - tx.amount.value
+                val dst = balances.getOrElse(tx.destination, Balance.empty.value.value) + tx.amount.value
+
+                balances
+                  .updated(tx.source, src)
+                  .updated(tx.destination, dst)
+            }
+            .toList
+            .foldLeftM(SortedMap.empty[Address, Balance]) {
```

Comment:
I wonder if we could come up with an implementation that only transforms addresses influenced by fee transactions here to balance, rather than all the balances (I guess most of them unchanged). Maybe we could start with finding all the src and dst addresses, then fetch balances for those addresses and then overwrite their balances in the balances map rather than taking the whole balances map in line 189.